synapse/python_dependencies.py

```diff
@@ -65,9 +65,7 @@
     "msgpack>=0.5.2",
     "phonenumbers>=8.2.0",
     "six>=1.10",
-    # prometheus_client 0.4.0 changed the format of counter metrics
-    # (cf https://github.com/matrix-org/synapse/issues/4001)
-    "prometheus_client>=0.0.18,<0.4.0",
+    "prometheus_client>=0.4.0,<0.8.0",
```

Comment:
actually: do we actually need 0.4.0? What happens on earlier versions?

Reply:
@richvdh 0.4.0 is where we need the customised exposition code, and it felt simpler to just have that than fall back to the old code on pre-0.4.0 and expand the number of versions there might be problems on, is all

Reply:
Right, but can't we use the customised code on 0.3, even though it's unnecessary? I'm just thinking of distros which currently have 0.3 or so.

Reply:
I... think so? let me check.

do we want 0.3 to be the bottom bound?

Reply:
> do we want 0.3 to be the bottom bound?

 well no, if it still works on 0.0.18, let's keep it at that.

Reply:
or rather: if we have no reason to believe that it doesn't work on 0.0.18, let's keep it at that. I'm not necessarily suggesting testing it.

Reply:
done